src/main/java/de/hysky/skyblocker/mixin/MinecraftClientMixin.java

```diff
@@ -22,4 +41,25 @@ public abstract class MinecraftClientMixin {
             HotbarSlotLock.handleInputEvents(player);
         }
     }
+
+    //Remove Downloading Terrain Screen and Reconfiguring Screen
+    @Inject(at = @At("HEAD"), method = "setScreen", cancellable = true)
+    public void setScreen(final Screen screen, final CallbackInfo ci) {
+        if (Utils.isOnSkyblock()) {
+            if (screen instanceof DownloadingTerrainScreen) {
+                ci.cancel();
+                this.setScreen(null);
+            } else if (screen instanceof ReconfiguringScreen && this.getNetworkHandler() != null) {
+                ci.cancel();
+                this.setScreen(new ReconfiguringPlaceholderScreen(this.getNetworkHandler().getConnection()));
+            }
+        }
+    }
+
+    @Redirect(method = "joinWorld", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/MinecraftClient;reset(Lnet/minecraft/client/gui/screen/Screen;)V"))
+    private void joinWorld(final MinecraftClient minecraft, final Screen screen) {
+        if (Utils.isOnSkyblock()) {
+            this.reset(new JoinWorldPlaceholderScreen());
+        }
+    }
```

Comment:
This removes the `reset` call when outside of skyblock. Maybe use a `ModifyArg` on the screen?
<img width="810" alt="Screenshot 2023-12-24 at 10 44 54" src="https://github.com/SkyblockerMod/Skyblocker/assets/92656833/35482061-e9c5-4136-9426-2444db3f7ab3">